### ISSUE TITLE
Decide what should be in the copyright header

### ISSUE THREAD

**Issue by OndrejM** [eclipse/microprofile-config#73]
This shouldn't be only config concern, but I'll raise it here as an issue.

We need to ask at Eclipse incubation what is the correct way to attribute the Copyright in the header, to avoid listing all parties in the header.

I know it was discussed in the mailing list, where Wayne provided some answers, but it is not clear yet. We should finally release a guideline and share it with everybody.

One solution suggested by Mark Struberg is to do something similar that Spring does: attribute to the main copyright holder and "other authors", where other authors are listed with @author javadoc tag (including full name and optionally their company).

The main concern is that whether person, his company, or even both could hold the copyright (often there are no legal reasons that an employee looses the copyright in favor of her employer). In case of external contractors, it makes even more sense to attribute copyright to him than to the company she works for, but it also makes sense to attribute copyright to both.

@kwsutter, would you ask for clarification and guidance at the incubation mailing list, or any Eclipse IP mailing list if it exists ?

**Comment by keilw** [comment]
This ("original authors") was tried by SmartHome recently, but @waynebeaton said on the mailing list before it was not feasible for Eclipse projects. And of all the projects under https://github.com/eclipse I could only think of two right now with this header approach. vert.x was originally created by people then working at SpringSource (now Pivotal) so it looks like that history gave it headers that look like Spring and it was never changed (it's also seemingly the only Eclipse project not using org.eclipse)

**Comment by struberg** [comment]
@keilw just because it 'used to be that way' doesn't make it more correct. In fact in many cases the Copyright header is **plain wrong** from a purely legal pov. Because one can only claim copyright if one invented something himself. Especially the need of a certain level of originality is simply not met by just copying over a class from another project to Microprofile. 
Every court in the world would deny such a copyright claim!

**Comment by keilw** [comment]
@struberg Every other Eclipse project does it, they focus on doing work instead of beating a dead horse with some pseudo-legal nonsense. The initial owner who started a module is always listed and there is a detailed contribution history in most correct Eclipse project sources. Eclipse IP team is used to checking what may be copied over or not. I know that first hand from several open CQs how much time and effort they spend on that ;-)

**Comment by keilw** [comment]
Sorry but this is not a blocker, the random  <groupId>org.eclipse.microprofile.apis</groupId>
is.

**Comment by struberg** [comment]
having wrong Copyrights in files is NOT a blocker? Interesting approach...

**Comment by keilw** [comment]
Eclipse has an IP team, if anything was wrong with a license header, THEY will tell, so unless you see a push being prevented or hear something (e.g. during release review) they claim is wrong, there is NO problem. The combination of "leading company or person" and a detailed commit history worked for Hundreds of Eclipse projects almost 20 years now. I think as a whole MicroProfile has so many different things to worry about.

**Comment by keilw** [comment]
Btw, isn't this redundant or a copy of the other one? Maybe @OndrejM or somebody else could close ONE of them, it seems more than enough to discuss a workable solution for these headers in one thread.

**Comment by OndrejM** [comment]
If we have an agreement in #45, I will close this ticket too and create a new one on a higher level to have a come to an agreement across all specs. Until then, I'll leave this issue open to remember to discuss also outside the config spec.

**Comment by keilw** [comment]
Unfortunately there is no agreement although it's only ranting and nagging, not even a distinct "+1", "-1" or similar (like the guys who don't vote and later complain about the outcome;-) 

So far in these tickets I saw 2 +1 for @OndrejM's suggestion of a Red Hat inspired 
```
Copyright (c) 2016-2017 Individual 1, Company 2, and other authors mentioned by the @author tag.

@author A B
@author X Y
```
Or different depending on the key initial contributors to THIS (microprofile/eclipse) repository.

**Comment by keilw** [comment]
Oh and SmartHome also only has a small number of bundles where they accidentially wrote "Copyright by the respective authors", while a large portion looks like
```
 * Copyright (c) 1997, 2015 by ProSyst Software GmbH and others.
```

**Comment by Emily-Jiang** [comment]
See this [thread ](https://groups.google.com/forum/#!topic/microprofile/sRqsnPTkkKE), from John Ament. We should update our files to use that.

**Comment by keilw** [comment]
Once we do, wouldn't it be a good idea to verify with the License Header Plugin like it's done in https://github.com/eclipse/microprofile-samples? Just for those scared, it does NOT alter files, the build only breaks if a header does not comply. Optionally (in mp-samples this is available in a special profile) all headers can be updated automatically, but if done manually it does not have to be used here. Before headers were different, if there is a common pattern now, it's good to check (I believe Jetty also does with the same plugin)

**Comment by struberg** [comment]
@Emily-Jiang yes we should. Gonna change all the files where I'm listed as Copyright holder to this pattern. Please do the same for all files with C IBM.
